From 7edfa89834594ed46e2a907d9b5e8bcfe335c6e2 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 14:38:14 -0400
Subject: [ethernet] enic: cisco-enic calls skb_set_hash

Message-id: <1409668716-4814-3-git-send-email-sassmann@redhat.com>
Patchwork-id: 90629
O-Subject: [RHEL7.1 PATCH 02/24] net: cisco-enic calls skb_set_hash
Bugzilla: 1107934
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Tom Herbert <therbert@google.com>

Drivers should call skb_set_hash to set the hash and its type
in an skbuff.

Signed-off-by: Tom Herbert <therbert@google.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 3739acddcce7a8cf1c8a938b4f4cda79ff4b1b78)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/cisco/enic/enic_main.c b/drivers/net/ethernet/cisco/enic/enic_main.c
index ff78dfa..b740bfc 100644
--- a/drivers/net/ethernet/cisco/enic/enic_main.c
+++ b/drivers/net/ethernet/cisco/enic/enic_main.c
@@ -1036,11 +1036,12 @@ static void enic_rq_indicate_buf(struct vnic_rq *rq,
   skb->protocol = eth_type_trans(skb, netdev);
   skb_record_rx_queue(skb, q_number);
   if (netdev->features & NETIF_F_RXHASH) {
-   skb->rxhash = rss_hash;
-   if (rss_type & (NIC_CFG_RSS_HASH_TYPE_TCP_IPV6_EX |
-     NIC_CFG_RSS_HASH_TYPE_TCP_IPV6 |
-     NIC_CFG_RSS_HASH_TYPE_TCP_IPV4))
-    skb->l4_rxhash = true;
+   skb_set_hash(skb, rss_hash,
+         (rss_type &
+          (NIC_CFG_RSS_HASH_TYPE_TCP_IPV6_EX |
+           NIC_CFG_RSS_HASH_TYPE_TCP_IPV6 |
+           NIC_CFG_RSS_HASH_TYPE_TCP_IPV4)) ?
+         PKT_HASH_TYPE_L4 : PKT_HASH_TYPE_L3);
   }
 
   if ((netdev->features & NETIF_F_RXCSUM) && !csum_not_calc) {
-- 
1.7.1